tests/test_layers.py

```diff
@@ -63,3 +68,35 @@ def test_override_settings(self):
     message = {"type": "test.message"}
     await layer.send("test.channel", message)
     assert message == await layer.receive("test.channel")
+
+
+@pytest.mark.parametrize(
+    "group_name,expected_valid",
+    [("¯\_(ツ)_/¯", False), ("chat", True), ("chat" * 100, False)],
+)
+def test_valid_group_name(group_name, expected_valid):
+    layer = InMemoryChannelLayer()
+    if expected_valid:
+        layer.valid_group_name(group_name)
+    else:
+        with pytest.raises(TypeError) as e:
+            layer.valid_group_name(group_name)
+            assert e.value.message.startswith("Group")
+            assert group_name in e.value.message
+            assert "< {}".format(MAX_NAME_LENGTH) in e.value.message
+
+
+@pytest.mark.parametrize(
+    "channel_name,expected_valid",
+    [("¯\_(ツ)_/¯", False), ("chat", True), ("chat" * 100, False)],
+)
+def test_valid_channel_name(channel_name, expected_valid):
+    layer = InMemoryChannelLayer()
+    if expected_valid:
+        layer.valid_channel_name(channel_name)
+    else:
+        with pytest.raises(TypeError) as e:
+            layer.valid_channel_name(channel_name)
```

Comment:
Shouldn't invalid test code be in test_invalid_channel_name ?

Reply:
The point is it's testing the function `valid_channel_name()`, hence `test_valid_channel_name()`.

Reply:
To reduce confusion and for merging tests for both methods, I'll change the test name regardless.